File: android/app/src/main/kotlin/com/wrbug/gitbbs/flutterbridge/FlutterBridge.kt
package com.wrbug.gitbbs.flutterbridge

import com.wrbug.gitbbs.flutterbridge.mmkv.MmkvFlutterChannel
import io.flutter.plugin.common.BinaryMessenger

class FlutterBridge(val messenger: BinaryMessenger) {
    private val channels: Array<FlutterChannel> by lazy {
        val arr: Array<FlutterChannel> = arrayOf(MmkvFlutterChannel())
        arr
    }


    fun inject() {
        channels.forEach {
            it.inject(messenger)
        }
    }
}

File: ios/Runner/FlutterBridge/FlutterBridge.swift
//
//  FlutterBridge.swift
//  Runner
//
//  Created by WrBug on 2019/3/7.
//  Copyright © 2019 The Chromium Authors. All rights reserved.
//

import Foundation

class FlutterBridge {
    var channels = [FlutterChannel]();
    init() {
        self.channels.append(MmkvFlutterChannel.init())
    }
    func inject( controller:FlutterViewController)  {
        channels.forEach { (channel) in
            channel.inject(controller: controller)
        }
    }
}
